include/Graph/Graph.hpp

```diff
@@ -653,6 +693,17 @@ const std::optional<const Edge<T> *> Graph<T>::getEdge(
   return std::nullopt;
 }
 
+template <typename T>
+std::set<Node<T> *> Graph<T>::nodeSet() {
+  std::set<Node<T> *> nodeSet;
+  for (auto &edgeSetIt : edgeSet) {
+    nodeSet.insert(const_cast<Node<T> *>(edgeSetIt->getNodePair().first));
```

Comment:
Same question, why this cast?

Reply:
I defined this private method because with getNodeSet I get a set of pointers to constant nodes, but in order to change the data attributes of the nodes I need pointers to non-const nodes

Reply:
For the question above I removed the cast because I didn't need it, since I was supposed to iterate over the set of non-const nodes. I fixed it.